project/src/components/movie-page/movie-page.tsx

```diff
@@ -1,18 +1,60 @@
-import { useParams } from 'react-router-dom';
-import { Film } from '../../types/film';
+import { useEffect } from 'react';
+import { connect, ConnectedProps } from 'react-redux';
+import { Link, useParams } from 'react-router-dom';
+import { AuthorizationStatus } from '../../const';
+import { fetchFilmAction, fetchReviewsAction, fethcSimilarFilmsAction } from '../../store/api-actions';
+import { ThunkAppDispatch } from '../../types/action';
+import { State } from '../../types/state';
 import FilmsList from '../films-list/films-list';
+import LoadingScreen from '../loading-screen/loading-screen';
 import Logo from '../logo/logo';
 import Tabs from '../tabs/tabs';
 import UserBlock from '../user-block/user-block';
 
-type MoviePageProps = {
-  moreLikeThisFilms: Film[];
-  films: Film[];
-}
+const mapStateToProps = ({ film, reviews, similarFilms, authorizationStatus }: State) => ({
+  film,
+  reviews,
+  similarFilms,
+  authorizationStatus,
+});
+
+const mapDispatchToProps = (dispatch: ThunkAppDispatch) => ({
+  fetchFilm(id: number) {
+    dispatch(fetchFilmAction(id));
+  },
+  fetchSimilarFilms(id: number) {
+    dispatch(fethcSimilarFilmsAction(id));
+  },
+  fetchReviews(id: number) {
+    dispatch(fetchReviewsAction(id));
+  },
+});
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
 
-function MoviePage({ moreLikeThisFilms, films }: MoviePageProps): JSX.Element {
+type PropsFromRedux = ConnectedProps<typeof connector>;
+type ConnectedComponentProps = PropsFromRedux;
+
+function MoviePage({ film, similarFilms, reviews, authorizationStatus, fetchFilm, fetchSimilarFilms, fetchReviews }: ConnectedComponentProps): JSX.Element {
```

Comment:
иногда лучше здесь описать просто `props` и вызывать дальше через точку `props.film` etc.
так можнг избежать таких длинных строк

```suggestion
function MoviePage(props: ConnectedComponentProps): JSX.Element {
```